crates/typos-cli/src/bin/typos-cli/main.rs

```diff
@@ -316,6 +317,11 @@ fn run_checks(args: &args::Args) -> proc_exit::ExitResult {
         if status_reporter.errors_found() {
             errors_found = true;
         }
+
+        if let Err(err) = status_reporter.generate_final_result() {
+            log::error!("generate final result: {}", err);
+            return proc_exit::Code::FAILURE.ok();
+        }
```

Comment:
This has us printing one SARIF json object per argument.

This should either be moved out of the loop or this is jsonlines and we shouldn't use `pretty`

Reply:
That's too bad I didn't notice that typos created a reporter for each file. i need to keep elevating the status of the reporter to make sure that the sarif file contains information from all the files.

Reply:
please take a look at https://github.com/crate-ci/typos/pull/1047/commits/7a834d04ec551250395aa1e72d8c59314e3f322c